src/App.js

```diff
@@ -1,23 +1,66 @@
-import logo from './logo.svg';
+
 import './App.css';
+import { useState } from 'react';
+import contacts from "./contacts.json"
+
+let contactsArray = contacts;
 
 function App() {
+  let contactsArr = [contacts[0], contacts[1], contacts[2], contacts[3], contacts[4]];
+  const[contactArr, setContacts] = useState(contactsArr)
+
+  const addRandomContact = ()=>{
+     setContacts(prevContacts=>{
+     const newList =  prevContacts.concat(contacts[Math.trunc(Math.random()*52) + 1]);
+      return newList;
+     })
+   }
+
+  const sortByAlphabet = ()=>{
+    setContacts(prevContacts=>{
+      const alphabeticList = prevContacts.sort((a,b)=>{
+        return a.name.localeCompare(b.name)
+      })
+      return alphabeticList;
+    })
+  }
```

Comment:
For this iteration, you firstly want to create a copy of the array you'll be working on, you do not really want to touch the original array of data - not to corrupt it. 

Otherwise you have correct solution here! 

For example:

```
const sortByAlphabet = ()=>{
    setContacts(prevContacts=>{
      const alphabeticList = [...prevContacts].sort((a,b)=>{
        return a.name.localeCompare(b.name)
      })
      return alphabeticList;
    })
  }
```